### ISSUE TITLE
Bug in PCD compression

### ISSUE THREAD

**Issue by taketwo** [PointCloudLibrary/pcl#2261]
## Your Environment
* Operating System and version: Ubuntu 16.04
* Compiler: gcc-5.4
* PCL Version: master

I was converting a dataset of non-compressed PCD files into binary compressed format to reduce space usage. While the conversion worked fine for most of the files, one output PCD file was corrupted. Attempting to open it results in:

> [pcl::PCDReader::read] Size of decompressed lzf data (4915197) does not match value stored in PCD header (4915200). Errno: 11

## How to reproduce

Download attached [cloud](https://github.com/PointCloudLibrary/pcl/files/1842539/cloud_00000016.pcd.zip), run it through the converter tool:

```bash
pcl_convert_pcd_ascii_binary cloud_00000016.pcd cloud_compressed.pcd 2
```

and try to open with the viewer.

## Possible solution

Did not have time to look into that yet.

**Comment by manhha1402** [comment]
I think I had the same problem when trying to save pcd files from my realsense camera. The number of points should be 307200(640*480), but it only saved 307199 points, I am not able to view pcd file in pcl_viewer

**Comment by taketwo** [comment]
Was a bug in *de*compression, solved by #2281.
